### PR TITLE
chore: add git hooks with husky

### PR DESCRIPTION
**Pull request by Thomaash**
- Husky to trigger them.
- Runs tests before every push.
- Lint Staged reformats changed code before every commit (honors .eslintignore).
- Commit Lint enforces conventional style of commit messages (another small step toward semantic-release).

### CODE REVIEW

File: test/data-view-chained.test.ts [hunk 1]
```diff
@@ -42,14 +42,29 @@ describe('Data view chaining', function(): void {
...
+        0,
+        1,
+        2,
+        3,
+        4,
+        5,
+        6,
+        7,
+        8,
+        9,
```
Comment (mojoaxel) [inline test/data-view-chained.test.ts]: well, this looks just stupid... :crab:

**Comment by mojoaxel** [comment]
@Thomaash I'm not sure what you changed but I personally like the new code style MUCH more than before, with the exception of these few change requests.

**Comment by Thomaash** [comment]
This is an error, there should be no reformatting in this PR. I'll have to look into this.

**Comment by Thomaash** [comment]
5 files, nice and small, only the stuff I wanted.

File: .huskyrc.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
[the remark this answers is not in the document]
Comment (Thomaash) [reply .huskyrc.json]: Done, I'll do it in the others too.

**Pull request merged**